onnx/numpy_helper.py

```diff
@@ -411,3 +411,16 @@ def convert_endian(tensor: TensorProto) -> None:
     tensor.raw_data = (
         np.frombuffer(tensor.raw_data, dtype=np_dtype).byteswap().tobytes()
     )
+
+
+def create_random_int_input_for_cast(
+    input_shape: Optional[Sequence[Union[str, int, None]]], cast_type: np.dtype
+) -> np.ndarray:
+    if cast_type not in (np.uint8, np.uint16, np.uint32, np.uint64):
```

Comment:
For creating int16 input, it is relatively safe to cast a float to an integer so typically the current test case will simply use `np.random.randn()` to create the input for testing. Since it would be just one line NumPy code, it seems fine not to include them in this function? This function is particularly handing the range of random integer creation for casting to unsigned integer.

Reply:
It's your choice. But I personally feel it is simpler to support it, to simplify future PRs ... because if somebody new to the codebase is introducing a test-case they are likely to copy-paste, and they may not be aware of this distinction. They could end up copying the `np.random.randn()` and use it for unsigned or vice-versa.

Reply:
After offline discussion, I also think it's good to add. I have made create_random_int support signed integer as well. PTAL. Thanks!